test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -1181,6 +1181,9 @@ public void OneHotHashEncodingOnnxConversionTest()
             var onnxModelPath = GetOutputPath(onnxFileName);
             SaveOnnxModel(onnxModel, onnxModelPath, null);
 
+            //Free up memory for x86 test
+            GC.Collect();
```

Comment:
Is this necessary? And how does this affect code coverage?

Reply:
I come up with out of memory issue for x86 in local env after upgrade package "Microsoft.NET.Test.Sdk".
So I'm assuming x86 test is close to use up all 2 GB memory. We should optimize memory usage like free up memory as soon as possible but that seems like out of this PR's scope.

---
In reply to: [432260986](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432260986) [](ancestors = 432260986)

Reply:
Are we running code coverage on x86 builds? If not, we should remove this from this PR and address it separately.

---
In reply to: [432322296](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432322296) [](ancestors = 432322296,432260986)

Reply:
breast-cancer.txt is a 15K file. OneHotHashEncoding is just doing a bunch of computation and not allocating memory. It is very suspicious that it would run out of memory during this test. Are you able to reproduce this consistently?

---
In reply to: [432724322](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432724322) [](ancestors = 432724322,432630210,432322296,432260986)

Reply:
Yes, I have several repro at local, haven't try on CI. Memory issue may not be at this test, what I observe is that when we start to see out of memory issue at local, we see this issue from several different tests but we always starts to see out of memory from this specific test. This is can be caused from previous code/test not releasing memory in time.

---
In reply to: [432725851](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432725851) [](ancestors = 432725851,432724322,432630210,432322296,432260986)

Reply:
Then it is not appropriate to add a GC.Collect here. Ideally we shouldn't be adding a GC.Collect anywhere and instead rely on predictable disposal of allocated memory. But if we have to add it, we should add it where the allocation is happening instead of somewhere else.

---
In reply to: [432738561](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432738561) [](ancestors = 432738561,432725851,432724322,432630210,432322296,432260986)

Reply:
Let me try on CI without this GC.Collect, if CI can pass then we can optimize memory issue in separate PR, other wise I would like to optimize memory usage first or disable some test from running on x86 first

---
In reply to: [432739891](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432739891) [](ancestors = 432739891,432738561,432725851,432724322,432630210,432322296,432260986)

Reply:
You can try adding memory statistics to BaseTestClass.Cleanup so that you can get a log of the memory usage at the end of each test to try and find the offending test. 

---
In reply to: [432761446](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432761446) [](ancestors = 432761446,432739891,432738561,432725851,432724322,432630210,432322296,432260986)

Reply:
That is great idea, I will do that in a separate PR since x86 test runs fine on CI due to my test

---
In reply to: [432787073](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432787073) [](ancestors = 432787073,432761446,432739891,432738561,432725851,432724322,432630210,432322296,432260986)